### PR TITLE
Add tag for typespec next

### PR DESCRIPTION
**Pull request by pshao25**
**Goal:**
Filter out all the typespec next run from https://dev.azure.com/azure-sdk/internal/_build?definitionId=5990

**Solution:**
For all the typespec next run , we add a tag `typespec_next` then we could choose from right top as below image shows. Or just this link: https://dev.azure.com/azure-sdk/internal/_build?definitionId=5990&_a=summary&tagFilter=typespec_next

image

### CODE REVIEW

File: eng/pipelines/preview.yml [hunk 1]
```diff
@@ -68,6 +68,13 @@ extends:
         $newContent | Set-Content $path -NoNewline
       displayName: 'Update Packages.Data.props'
       workingDirectory: $(sdkRepositoryPath)
+    AutorestInitializationSteps:
+    - ${{ if eq(parameters.UseTypeSpecNext, 'true') }}:
+      - task: Bash@3
+        inputs:
+          targetType: 'inline'
+          script: 'echo "##vso[build.addbuildtag]typespec_next"'
```
Comment (archerzz) [inline eng/pipelines/preview.yml:76]: so the magic bash script can add a filter in the UI? Is this the formal way to do so?

Comment (pshao25) [reply eng/pipelines/preview.yml:76]: > so the magic bash script can add a filter in the UI?

Yes.

> Is this the formal way to do so?

To be honest I don't know the answer. But we don't have other ways.

Comment (pshao25) [reply eng/pipelines/preview.yml:76]: @hallipr Could you help review this PR?

**Review by lirenhe — APPROVED** [review]
This is a workable solution so I approved this one.  @m-nash, could you also help to review this PR?  thanks

File: eng/pipelines/preview.yml (hunk 1 above)
Comment (hallipr) [reply eng/pipelines/preview.yml:76]: This is the standard way to add tags to a build mid-run.  See:

[Scheduled Builds](https://github.com/Azure/azure-sdk-for-net/blob/Azure.Core_1.40.0/eng/pipelines/templates/jobs/ci.yml#L72)

[Release Tags](
https://github.com/Azure/azure-sdk-for-net/blob/Azure.Core_1.40.0/eng/pipelines/templates/jobs/ci.mgmt.yml#L54)

Comment (hallipr) [inline eng/pipelines/preview.yml:76]: I would use powershell for xplat support

```suggestion
      - pwsh: |
          echo "##vso[build.addbuildtag]typespec_next"
```

**Review by hallipr — APPROVED** [review]

**Pull request merged**